istioctl/cmd/workload.go

```diff
@@ -390,11 +391,25 @@ func createMeshConfig(kubeClient kube.ExtendedClient, wg *clientv1alpha3.Workloa
 		md["ISTIO_META_AUTO_REGISTER_GROUP"] = wg.Name
 	}
 
-	proxyYAML, err := gogoprotomarshal.ToYAML(meshConfig.DefaultConfig)
+	proxyConfig, err := gogoprotomarshal.ToJSONMap(meshConfig.DefaultConfig)
 	if err != nil {
 		return err
 	}
-	return ioutil.WriteFile(filepath.Join(dir, "mesh.yaml"), []byte(proxyYAML), filePerms)
+
+	proxyYAML, err := yaml.Marshal(map[string]interface{}{"defaultConfig": proxyConfig})
```

Comment:
Is this going to be a backwards incompatible change?

Reply:
oh nevermind, its already broken today?

Reply:
Yeah, this was broken to begin with.

Reply:
https://github.com/istio/istio/issues/29125 mentions that the generated mesh config isn't properly formatted.